external/kafka/src/test/scala/org/apache/spark/streaming/kafka/KafkaRDDSuite.scala

```diff
@@ -40,43 +41,70 @@ class KafkaRDDSuite extends KafkaStreamSuiteBase with BeforeAndAfter {
     tearDownKafka()
   }
```

Comment:
Sure, but I think there are two bigger things that could be done here

1.  Change the thread sleep in setupKafka to polling using eventually

2.  Change this group of tests to use nested suites, so kafka / zk setup is only done once